rewards-squid/src/blocks/utils.ts

```diff
@@ -0,0 +1,117 @@
+import { Account, Nominator, Operator } from "../model";
+import { ProcessorContext } from "../processor";
+import { Store } from '@subsquid/typeorm-store';
+import { ApiPromise } from "@polkadot/api";
+
+/**
+ *
+ * @param ctx
+ * @param api
+ * @returns
+ */
+  export async function getOrCreateOperator(
+    ctx: ProcessorContext<Store>,
+    operatorId: bigint,
+    blockNumber: number,
+    api: ApiPromise
+  ): Promise<Operator | undefined> {
+    let operator = await ctx.store.get(Operator, operatorId.toString());
+
+    const nominators = await api.query.domains.nominators.entries(operatorId);
+    const nominatorsLength = nominators.length;
+
+    if (!operator) {
+      const operatorInfo = (
+        await api.query.domains.operators(operatorId)
+      ).toJSON() as any;
+
+      const ownerAccount = (
+        await api.query.domains.operatorIdOwner(operatorId)
+      ).toJSON();
+      if (operatorInfo) {
+        operator = new Operator({
+          id: operatorId.toString(),
+          orderingId: Number(operatorId),
+          operatorOwner: ownerAccount?.toString(),
+          status: operatorInfo.status.toString(),
+          signingKey: operatorInfo.signingKey,
+          totalShares: BigInt(operatorInfo.totalShares),
+          currentEpochRewards: operatorInfo.currentEpochRewards,
+          currentTotalStake: BigInt(operatorInfo.currentTotalStake),
+          nominatorAmount: nominatorsLength,
+          nominationTax: operatorInfo.nominationTax,
+          minimumNominatorStake: BigInt(operatorInfo.minimumNominatorStake),
+          nextDomainId: operatorInfo.nextDomainId,
+          currentDomainId: operatorInfo.currentDomainId,
+          updatedAt: BigInt(blockNumber),
+        });
+
+        await ctx.store.insert(operator);
+      }
+    } 
+
+    return operator;
+  };
+
+export async function getOrCreateNominators(
```

Comment:
each nominator stake and share should be added to a aggregate total in the operator object...

Reply:
The operator has already added stake and share values in its own object. So, it's redundant to repeat them.